Weather Man/weather_reports.py

```diff
@@ -0,0 +1,97 @@
+    - generate_daily_temperatures_report(year, month): Generates a daily temperatures report for a specific year and month.
+    """
+
+    def __init__(self, calculation_results):
+        """
+        Initializes WeatherReport with calculation results.
+
+        Args:
+        - calculation_results: Calculation results object containing weather data.
+        """
+
+        self.results = calculation_results
+
+    def generate_extreme_weather_yearly_report(self):
+        """
+        Generates a report for highest temperature, lowest temperature, and humidity extremes.
+        Returns:
+        - String containing the report.
+        """
+        output = ""
+
+        if (self.results["highest_temperature"]
+                and self.results["lowest_temperature"]
+                and self.results["most_humid_day"]):
+            highest_temp, highest_day = self.results["highest_temperature"]
+            lowest_temp, lowest_day = self.results["lowest_temperature"]
+            humidity, humid_day = self.results["most_humid_day"]
+            output = (
+                f"Highest: {round(highest_temp, 1)}C on {highest_day.strftime("%B")} {highest_day.strftime("%d")}\n"
+                f"Lowest: {round(lowest_temp, 1)}C on {lowest_day.strftime("%B")} {lowest_day.strftime("%d")}\n"
+                f"Humidity: {round(humidity, 1)}% on {humid_day.strftime("%B")} {humid_day.strftime("%d")}"
+                )
+        else:
+            output = "No data available for extremes for given input."
+        return output
+
+    def generate_average_weather_yearly_report(self):
+        """
+        Generates a report for average highest temperature, average lowest temperature, and average mean humidity.
+
+        """
+        output = ""
+        if (self.results["average_highest_temperature"]
+                and self.results["average_lowest_temperature"]
+                and self.results["average_mean_humidity"]):
+
+            output = (f"Average highest: {round(self.results['average_highest_temperature'], 1)}C\n"
+                      f"Average lowest: {round(self.results['average_lowest_temperature'], 1)}C\n"
+                      f"Average humidity: {round(self.results['average_mean_humidity'], 1)}%")
+        else:
+            output = "No data available for averages for given input."
+        return output
+
+    def generate_daily_temperatures_report(self, year, month):
+        """
+        Generates a daily temperatures report for a specific year and month.
+
+        Args:
+        - year (int): Year for which the report is generated.
+        - month (int): Month (1-12) for which the report is generated.
+        """
+
+        month_name = datetime(year, month, 1).strftime('%B')
+        output = ""
+        for date, max_temp, min_temp in self.results["daily_temperatures"]:
+            day = date.strftime('%d')
+            blue_stars = print_colored_stars(min_temp, "blue")
```

Comment:
make the function in a way that you pass min and max temperature. and it provides the output